### ISSUE TITLE
all Data struct is static

### ISSUE THREAD

**Issue by mzprog** [cvhariharan/Tiny-Basic#4]
after reading and studying the code i noticed that struct all are static , i prefer to change to something more flexible<br>
like: 
`Token` , `Symbol`,  `stack`,
all needed to be in linked lists

some functions should be turned to pointer lilke:
`Token eat(int type)`  --> `Token * eat(int type)`
because on failed no way to know it

do you have any different idea or start do it?

**Comment by cvhariharan** [comment]
Hi, I agree that using a dynamic type to hold tokens and symbols is the right way to do this but back when I wrote this, I wanted to write something small enough that even a beginner in C could understand. Adding linked lists or ASTs would increase the code size. I wrote this mainly to learn about interpreters and never really tried to make it efficient or polished.
Having said that, modifying certain functions to return pointers could later help to add error handling, which is currently absent. 
Hope this helps. Cheers!
